libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,507 @@
+        }
+        else if( TransportSocketStatus < ( int32_t ) 0 )
+        {
+            LogError( ( "Error %ld while receiving data.", TransportSocketStatus ) );
+        }
+        else
+        {
+            bytesReceived = ( size_t ) TransportSocketStatus;
+        }
+
+        if( bytesReceived < bytesToRecv )
+        {
+            LogInfo( ( "Receive requested %lu bytes, but %lu bytes received instead.",
+                       bytesToRecv,
+                       bytesReceived ) );
+        }
+        else
+        {
+            LogInfo( ( "Successfully received %lu bytes.",
+                       bytesReceived ) );
+        }
+    }
+
+    return ( int32_t ) bytesReceived;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t tlsSetup( const SocketsConfig_t * pSocketsConfig,
+                         Socket_t tcpSocket,
+                         const uint8_t * pHostName,
+                         size_t hostnameLength )
+{
+    int32_t TransportSocketStatus = SOCKETS_ERROR_NONE;
+
+    /* ALPN options for AWS IoT. */
+    /* ppcALPNProtos is unused. putting here to align behavior in IotNetworkAfr_Create. */
+    /* coverity[misra_c_2012_rule_4_6_violation] */
+    /* coverity[misra_c_2012_rule_8_13_violation] */
+    const char * ppcALPNProtos[] = { socketsAWS_IOT_ALPN_MQTT };
+
+    /* Set secured option. */
+    TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                0,
+                                                SOCKETS_SO_REQUIRE_TLS,
+                                                NULL,
+                                                0 );
+
+    if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+    {
+        LogError( ( "Failed to set secured option for soocket." ) );
+    }
+
+    /* Set ALPN option. */
+    if( ( TransportSocketStatus == SOCKETS_ERROR_NONE ) && ( pSocketsConfig->pAlpnProtos != NULL ) )
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_ALPN_PROTOCOLS,
+                                                    ppcALPNProtos,
+                                                    sizeof( ppcALPNProtos ) / sizeof( ppcALPNProtos[ 0 ] ) );
+
+        if( TransportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+        {
+            LogError( ( "Failed to set ALPN option socket." ) );
+        }
+    }
+
+    /* Set SNI option. */
+    if( ( TransportSocketStatus == SOCKETS_ERROR_NONE ) && ( pSocketsConfig->disableSni == false ) )
+    {
+        TransportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                    0,
+                                                    SOCKETS_SO_SERVER_NAME_INDICATION,
+                                                    pHostName,
+                                                    hostnameLength + 1 );
```

Comment:
Is the NULL character required in the length parameter?

Reply:
This is following 
https://github.com/aws/amazon-freertos/blob/master/libraries/abstractions/platform/freertos/iot_network_freertos.c#L275

Added a description to mention that is requires to be null terminated for pHostName in 30845196c6f16ed3981ed970a1f4cd47c761da77